lib/vscode/build/tsconfig.json

```diff
@@ -15,7 +15,8 @@
 		"strict": true,
 		"noUnusedLocals": true,
 		"noUnusedParameters": true,
-		"newLine": "lf"
+		"newLine": "lf",
+		"skipLibCheck": true
```

Comment:
I added this to prevent TypeScript from checking the declaration files at build time. This resolves the @types issue with mocha and jest using the same “test” namespace in their declaration files. According to the handbook:
> Skip type checking of declaration files.
> This can save time during compilation at the expense of type-system accuracy

The recommended setting is `true`. I think this is safe to change but appreciate input from y’all @nhooyr @code-asher 

Suggested reading:
- https://github.com/DefinitelyTyped/DefinitelyTyped/issues/43912
- [TSConfig Reference: skipLibCheck](https://www.typescriptlang.org/tsconfig#skipLibCheck)

Reply:
Doesn't this mean we have no type checking now or does it mean we only get type checking when linting?

Reply:
Ah sorry I see. Yea I don't think we can do this, we rely on declaration files throughout and I'm sure Microsoft does as well.

Reply:
Has this been fixed @jsjoeio?

Reply:
Discussed on Slack! I will revert to draft, fix CI, then mark ready and re-request a review

Reply:
Discussed again and it looks like we can't do anything better than skipLibCheck until https://github.com/microsoft/TypeScript/issues/37053 is resolved by the tsc team.